src/libraries/Microsoft.Extensions.Logging/src/Logger.cs

```diff
@@ -219,6 +219,11 @@ private sealed class LoggerProviderDebugView(string providerName, MessageLogger?
                     return null;
                 }
 
+                if (!logger.Value.MinLevel.HasValue)
```

Comment:
I don't think this is the right fix. 

The fix should change the line 181:
```C#
MessageLogger? messageLogger = logger.MessageLoggers?.FirstOrDefault(messageLogger => messageLogger.Logger == loggerInfo.Logger);
```

To:

```C#
            MessageLogger? messageLogger = FirstOrNull(logger.MessageLoggers, loggerInfo.Logger);

            MessageLogger? FirstOrNull(MessageLogger[]? array, ILogger logger)
            {
                if (array is null || array.Length == 0)
                {
                    return null;
                }

                foreach (var item in array)
                {
                    if (item.Logger == logger)
                    {
                        return item;
                    }
                }

                return null;
            }

```

Reply:
To give some more details, in the failing example, `logger.MessageLoggers` will be empty array. The line:

```C#
MessageLogger? messageLogger = logger.MessageLoggers?.FirstOrDefault(messageLogger => messageLogger.Logger == loggerInfo.Logger);
```

will return `messageLogger` equal to the `default`. `MessageLogger` is a value type so the returned value will never be null and always has HasValue = true. Then we do `new LoggerProviderDebugView(providerName, messageLogger)` passing the default value. The `LoggerProviderDebugView.CalculateEnabledLogLevel` will do the check:

```C#
                if (logger == null)
                {
                    return null;
                }

```

which will never be true.

Reply:
CC @JamesNK

Reply:
Thank you @tarekgh. I didn't notice that `MessageLogger` was a struct, so `FirstOrDefault` won't return null. I applied and tested your suggestion.

Reply:
You could also do this:

```cs
MessageLogger? messageLogger = logger
    .MessageLoggers
    ?.Cast<MessageLogger?>()
    .FirstOrDefault(messageLogger => messageLogger.Logger == loggerInfo.Logger);
```

But adding the new method is fine.